File: java/src/test/java/com/kcl/ValidateTest.java
package com.kcl;

import org.junit.Assert;
import org.junit.Test;

import com.kcl.api.API;
import com.kcl.api.Spec.ValidateCodeArgs;
import com.kcl.api.Spec.ValidateCodeResult;

public class ValidateTest {
    @Test
    public void testValidateCodeApi() throws Exception {
        String code = "schema Person:\n" + "    name: str\n" + "    age: int\n" + "    check:\n"
                + "        0 < age < 120\n";

        String data = "{\"name\": \"Alice\", \"age\": 10}";

        ValidateCodeArgs args = ValidateCodeArgs.newBuilder().setCode(code).setData(data).setFormat("json").build();
        API apiInstance = new API();
        ValidateCodeResult result = apiInstance.validateCode(args);

        Assert.assertTrue(result.getSuccess());
        Assert.assertEquals("", result.getErrMessage());
    }
}

File: kotlin/src/test/kotlin/com/kcl/ValidateTest.kt
package com.kcl;

import com.kcl.api.API
import com.kcl.api.validateCodeArgs
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.Assertions.assertEquals

class ValidateTest {
    @Test
    fun testValidateCodeApi() {
        val args = validateCodeArgs { 
            code =  "schema Person:\n" + "    name: str\n" + "    age: int\n" + "    check:\n" + "        0 < age < 120\n"
            data = "{\"name\": \"Alice\", \"age\": 10}"
        }
        val apiInstance = API();
        val result = apiInstance.validateCode(args);

        assertEquals(true, result.getSuccess());
        assertEquals("", result.getErrMessage());
    }
}
